packages/xo-web/src/xo-app/home/index.js

```diff
@@ -448,6 +450,14 @@ const NoObjects = props =>
 })
 @connectStore(() => {
   const type = (_, props) => props.location.query.t || DEFAULT_TYPE
+  const nItemsPerPage = (_, props) => {
+    let nItems
+    if ((nItems = cookies.get('nItemsPerPage')) !== undefined) {
+      return nItems
+    }
+    nItems = +props.location.query.n
```

Comment:
This suggestion add an additional condition but it simplify the code
```js
const nItems = defined(cookies.get('nItemsPerPage'), +props.location.query.n)
```

Reply:
IMHO, the query string should take precedence over cookies.

Reply:
Why are we using the query string and the cookies at the same time?

Reply:
Don't know, but if we do, cookies should take precedence.

Reply:
```js
(_, props) => +defined(
  props.location.query.n,
  () => cookies.get('nItemsPerPage'),
  DEFAULT_ITEMS_PER_PAGE
)
```